src/infra/postgres/client/client.go

```diff
@@ -80,19 +84,25 @@ func (c *PostgresClient) SelectDeleted(ctx context.Context, model ...interface{}
 }
 
 func (c *PostgresClient) UpdatePK(ctx context.Context, model ...interface{}) error {
-	_, err := c.db.ModelContext(ctx, model...).WherePK().Update()
+	res, err := c.db.ModelContext(ctx, model...).WherePK().Update()
 	if err != nil {
 		return parseErrorResponse(err)
 	}
+	if res.RowsAffected() != 1 {
+		return errors.NotFoundError("update not effected")
+	}
 
 	return nil
 }
 
 func (c *PostgresClient) DeletePK(ctx context.Context, model ...interface{}) error {
-	_, err := c.db.ModelContext(ctx, model...).WherePK().Delete()
+	res, err := c.db.ModelContext(ctx, model...).WherePK().Delete()
 	if err != nil {
 		return parseErrorResponse(err)
 	}
+	if res.RowsAffected() != 1 {
```

Comment:
I doubt you need this, if the data is not found, you should get a `err == pg.ErrNoRows`. The data cannot be found but nor deleted, otherwise you get an error. You can simply remove this `if` IMO.